File: src/main/java/com/yanns/java/example1/version1/InvoiceController.java
package com.yanns.java.example1.version1;

import java.util.Date;

public class InvoiceController {

    InvoiceCreator invoiceCreator = new InvoiceCreator();

    public void payInvoice() {
        String newInvoiceId = invoiceCreator.createInvoiceName(new Date(), "45AE-45F", "872346", "abc-234", true, false, false);

        // ...
    }
}

File: src/main/scala/com/yanns/scala/example1/version1/InvoiceController.scala
package com.yanns.scala.example1.version1

import java.util.Date

class InvoiceController {

  val invoiceCreator = new InvoiceCreator()

  def payInvoice {
    val newInvoiceId = invoiceCreator.createInvoiceName(new Date, "45AE-45F", "872346", "abc-234", true, false, false)

    // ...
  }

}
